packages/safe-tools-client/app/components/future-payments-list/index.gts

```diff
@@ -1,49 +1,110 @@
+  Args: {
+    onDepositClick: () => void;
+  }
+}
+
+export default class FuturePaymentsList extends Component<Signature> {
   @service declare hubAuthentication: HubAuthenticationService;
   @service declare network: NetworkService;
   @service declare scheduledPayments: ScheduledPaymentsService;
-  @service declare tokens: TokensService;
-  @service declare wallet: WalletService;
 
-  get futurePayments() {
-    return [] //TODO: fetch future payments
+  get nextHour(): ScheduledPayment[] {
+    if (!this.futurePaymentsResource.value) return [];
+    return this.futurePaymentsResource.value.filter((s: ScheduledPayment) => s.payAt >= addHours(new Date(), 1) && s.payAt <= addHours(new Date(), 2));
+  }
+
+  get nextMonth(): ScheduledPayment[] {
+    if (!this.futurePaymentsResource.value) return [];
+    let now = new Date();
+    let nextMonth = addMonths(now, 1);
+    let firstDateNextMonth = new Date(nextMonth.getFullYear(), nextMonth.getMonth(), 1, 0, 0, 0, 0);
+    let lastDateNextMonth = new Date(nextMonth.getFullYear(), nextMonth.getMonth(), lastDayOfMonth(nextMonth).getDate(), 23, 59, 0, 0);
+    return this.futurePaymentsResource.value.filter((s: ScheduledPayment) => s.payAt >= firstDateNextMonth && s.payAt <= lastDateNextMonth);
+  }
+
+  get nextFewMonthsPayments(): ScheduledPayment[] {
+    if (!this.futurePaymentsResource.value) return [];
+    let now = new Date();
+    let nextMonth = addMonths(now, 1);
+    let lastDateNextMonth = new Date(nextMonth.getFullYear(), nextMonth.getMonth(), lastDayOfMonth(nextMonth).getDate(), 23, 59, 0, 0);
+    return this.futurePaymentsResource.value.filter((s: ScheduledPayment) => s.payAt > lastDateNextMonth);
   }
 
-  @action
-  async addFunds() {
-    //TODO: implement add fund action
+  @task *loadFuturePaymentTask(chainId: number): TaskGenerator<ScheduledPayment[]> {
+    return yield this.scheduledPayments.fetchScheduledPayments(chainId, new Date());
   }
 
+  @use futurePaymentsResource = resource(() => {
+    if (!this.hubAuthentication.isAuthenticated) {
+      return {
+        error: false,
+        isLoading: false,
+        value: [],
+      };
+    }
+
+    const state = new TrackedObject({
+      isLoading: false,
+      value: undefined as ScheduledPayment[] | undefined,
+      error: undefined,
+    });
+
+    let chainId = this.network.chainId;
+
+    (async () => {
+      try {
+        state.value = await taskFor(this.loadFuturePaymentTask).perform(chainId);
+      } catch (error) {
+        console.log(error);
+        state.error = error;
+      } finally {
+        state.isLoading = false;
+      }
+    })();
+
+    return state;
+  });
+
   <template>
-    <BoxelActionContainer
+    <BoxelActionContainer 
       class="future-payments-list"
       as |Section ActionChin|>
-      {{#if (lt this.futurePayments.length 1)}}
-        <Section class="future-payments-list__no-payments-section">
+      {{#if (lt this.futurePaymentsResource.value.length 1)}}
```

Comment:
This will briefly flicker with the "no payments found" text when the payments are loading initially. Make sure to handle this state and only after the loading is done decide if you need to show the "no payments found" screen